core/src/main/java/io/questdb/std/Files.java

```diff
@@ -138,10 +146,36 @@ public static long getOpenFileCount() {
 
     public native static long getStdOutFd();
 
+    public static boolean isDir(long pUtf8NameZ, long type, StringSink nameSink) {
+        if (type == DT_DIR) {
+            nameSink.clear();
+            Chars.utf8DecodeZ(pUtf8NameZ, nameSink);
+            return notDots(nameSink);
+        }
+        return false;
+    }
+
+    public static boolean isDir(long pUtf8NameZ, long type) {
+        return type == DT_DIR && notDots(pUtf8NameZ);
+    }
+
     public static boolean isDots(CharSequence name) {
         return Chars.equals(name, '.') || Chars.equals(name, "..");
     }
 
+    /**
+     * Detects if filesystem is supported by QuestDB. The function returns both FS magic and name. Both
+     * can be presented to user even if file system is not supported.
+     *
+     * @param lpszName existing path on the file system. The name of the filesystem is written to this
+     *                 address, therefore name should have at least 128 byte capacity
+     * @return 0 when OS call failed, errno should be checked. Negative number is file system magic that is supported
+     * positive number is magic that is not supported.
+     */
+    public static int getFileSystemStatus(LPSZ lpszName) {
+        return getFileSystemStatus(lpszName.address());
```

Comment:
I'd ensure lpszName has enough capacity to accept return string with an `assert` or `if`

Reply:
done here: https://github.com/questdb/questdb/pull/1725/commits/444239783f1a60cd1886777a8eb03f12e9b458cf